### DISCUSSION TITLE
Dirackernel oscillation

### DISCUSSION THREAD

**Discussion by Joseph-0123** [idaholab/moose#23777] [Q&A General] [answered]
Dear all,

I am using the dirackernel 'ConstantPointSource' in our moose-based [app](https://github.com/MGK-Lab/tiger) to simulate the injected fluid at one point on a 1D line. But the oscillation occurs. If the mesh is refined from 1m to 0.5, the oscillation will be weakened but not relatively smooth as shown in the [Numerical diffusion example](https://mooseframework.inl.gov/modules/porous_flow/numerical_diffusion.html) of the Porosflow module. Numerical stabilization matters here.

Are there any boundary conditions or other settings that can solve this kind of oscillation problem or replace the function of the dirackernels？Thanks a lot.

J

[DiracKernels]
  [./out_pp]
    type = ConstantPointSource
    point = '-1998 0 0'
    variable = poressure
    value = -400
  [../]
[]

Initial oscillation 
![image](https://user-images.githubusercontent.com/56687392/226124331-e33473b0-5e86-445d-95b0-d65904b5a3b0.png)

Weakened oscillation after refinement
![image](https://user-images.githubusercontent.com/56687392/226124489-7844fa7a-a35c-433f-b6c8-2f02223572ed.png)

**Comment by Joseph-0123** [comment]
Hello @GiudGiud , could you please help me with this? Thanks.

**Reply by GiudGiud** [reply, marked as answer]
Dirac kernels introduce a discontinuous term, it s not all that surprising to see oscillations. 

You can refine or smooth the source term with a volumetric source

if this is for porous flow we should re classify this in the porous flow QNA so the developers pitch in
